.github/actions/verify-token/src/utils.ts

```diff
@@ -22,3 +27,172 @@ export function getWorkflowPath(obj: githubObj): string {
     .substring(`${obj.repository}/`.length)
     .split("@", 1)[0];
 }
+
+export function asMap(
+  inputs: Map<string, string | number | boolean>
+): Map<string, string | number | boolean> {
+  if (!inputs) {
+    return new Map<string, string | number | boolean>();
+  }
+  if (!(inputs instanceof Map)) {
+    return new Map<string, string | number | boolean>(Object.entries(inputs));
+  }
+  return inputs;
+}
+
+export function parseCertificate(
+  bundle: sigstore.sigstore.Bundle
+): [string, string, string, string, string] {
+  if (bundle === undefined) {
+    throw new Error(`undefined bundle.`);
+  }
+  if (bundle.verificationMaterial === undefined) {
+    throw new Error(`undefined bundle.verificationMaterial.`);
+  }
+  if (bundle.verificationMaterial.x509CertificateChain === undefined) {
+    throw new Error(
+      `undefined bundle.verificationMaterial.x509CertificateChain.`
+    );
+  }
+  if (
+    bundle.verificationMaterial.x509CertificateChain.certificates.length === 0
+  ) {
+    throw new Error(
+      `bundle.verificationMaterial.x509CertificateChaincertificates is empty.`
+    );
+  }
+  // NOTE: the first certificate is the client certificate.
+  const clientCertDer = Buffer.from(
+    bundle.verificationMaterial.x509CertificateChain.certificates[0].rawBytes,
+    "base64"
+  );
+  const clientCertPath = "client.cert";
```

Comment:
nit: Maybe consider writing to `$RUNNER_TEMP` rather than the workspace?

Reply:
Added this comment https://github.com/slsa-framework/slsa-github-generator/issues/1918. Since this changes tscommon, I think we should resolve that issue first.